libdnf5/base/goal.cpp

```diff
@@ -596,6 +601,130 @@ GoalProblem Goal::Impl::add_module_specs_to_goal(base::Transaction & transaction
     return ret;
 }
 
+GoalProblem Goal::Impl::add_transaction_replay_specs_to_goal(base::Transaction & transaction) {
+    libdnf5::GoalJobSettings settings = transaction_replay_settings;
+    bool skip_unavailable = settings.resolve_skip_unavailable(base->get_config());
+
+    for (const auto & package_replay : transaction_replay.packages) {
+        libdnf5::GoalJobSettings settings_per_package = settings;
```

Comment:
May I ask you why you create a copy of settings here?

Reply:
The `settings` are used to select the repository which can be different for each action. I wanted to avoid the actions affecting each other.
But then it is copied again into the particular vector of specs so it would be possible to have the settings just once and make sure we always overwrite the `to_repo_ids` or other used values.

I can change it if you prefer that? It would be more efficient.

Reply:
Thank you for explanation. I am fine with that.